org.palladiosimulator.analyzer.slingshot.behavior.spd/src/org/palladiosimulator/analyzer/slingshot/behavior/spd/interpreter/entity/aggregator/AbstractWindowAggregation.java

```diff
@@ -0,0 +1,70 @@
+package org.palladiosimulator.analyzer.slingshot.behavior.spd.interpreter.entity.aggregator;
+
+import java.util.ArrayDeque;
+import java.util.Queue;
+
+/**
+ * A window-based aggregator with a window size of the last
+ * {@code windowSize} measurements.
+ * 
+ * @author Julijan Katic
+ */
+public abstract class AbstractWindowAggregation {
```

Comment:
If we follow Esper, http://esper.espertech.com/release-8.8.0/reference-esper/html/epl-views.html and the data windows, this seems like a FixedLengthWindowAggregation ;). In the future we could have TimeWindowAggregation or other types.